Production/python/TauL2Prod.py

```diff
@@ -0,0 +1,185 @@
+import FWCore.ParameterSet.Config as cms
+from FWCore.ParameterSet.VarParsing import VarParsing
+from Configuration.Eras.Era_Run3_cff import Run3
+import os
+
+# to keep sampleType, fileList, fileNamePrefix, tupleOutput, lumiFile, eventList and dumpPython
+
+options = VarParsing('analysis')
+options.register('sampleType', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "Indicates the sample type: Summer16MC, Run2016, ...")
+options.register('fileList', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "List of root files to process.")
+options.register('fileNamePrefix', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "Prefix to add to input file names.")
+options.register('tupleOutput', 'eventTuple.root', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "Event tuple file.")
+options.register('lumiFile', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "JSON file with lumi mask.")
+options.register('eventList', '', VarParsing.multiplicity.singleton, VarParsing.varType.string,
+                 "List of events to process.")
+options.register('dumpPython', False, VarParsing.multiplicity.singleton, VarParsing.varType.bool,
+                 "Dump full config into stdout.")
+options.parseArguments()
+
+isData = False
+if 'Data' in options.sampleType:
```

Comment:
check that sample type is among those that you expect 'Run3MC' or `Run2Data`, otherwise raise an error, because the config most certainly will not be valid for other sample types.